bin/node/runtime/src/lib.rs

```diff
@@ -849,14 +849,18 @@ impl pallet_society::Trait for Runtime {
 
 parameter_types! {
 	pub const MinVestedTransfer: Balance = 100 * DOLLARS;
+	// For weight estimation, we assume that the most locks on an individual account will be 50.
+	// This number may need to be adjusted in the future if this assumption no longer holds true.
+	pub const MaxLocks: u32 = 50;
```

Comment:
I don't think this makes much sense. Reading this, I assume that if I set this, then the maximum locks are limited to 50, but this is not the case. Setting this to 5 does not limit anything. 

If we actually need to have a limit on the number of locks, then if there is a way for it, it makes a lot more sense to put this config as `type` in system trait or as `const` in `Currency` trait. 

Another reason is that I am sure there will be (and already is) more modules that have operations on lock, and therefore their benchmarks will need some form of `MaxLocks`.

If there is no need for the system to make this assumption, then I think it is fine to just make a constant that is large enough and do the same `let l in 0 .. MAX_LOCKS;` If `MAX_LOCK` is large enough, then the formula won't change regardless of what `MAX_LOCK` is. 

---

Reply:
Having some hard limit in the balances module is hard because it would ultimately change the API around locks, where adding a lock may return a result, and thus any logic which adds a lock may need to now handle this error, and it could be quite an impact across the runtime.

As for "selecting a large number", that is basically what I have done here, except as I have learned more consistently over time, we should make this configurable. A "big number" for us may not be a big number for someone else. Or maybe our choice is crazy large for some other runtime.

Basically, all the functions in vesting are some `f(max_locks)`, and so we let them define those functions, and we also let them define the `max_locks` variable.

Reply:
maybe we can change the name to estimated_max_lock ? or weight_for_locks ?

but I agree that if vesting needs MaxLocks then some other pallet will need it as well.

Reply:
Updated per discussion. A new trait is introduced into the Balances Pallet and the `LockableCurrency` trait. And this is ingested in the pallet.